File: python/src/findDuplicate_test.py
from typing import Callable, List

import pytest

from findDuplicate import Solution

solution = Solution()


@pytest.mark.parametrize(
    "fn",
    (
        (solution.findDuplicate),
        (solution.findDuplicate_floyd),
    ),
)
@pytest.mark.parametrize(
    "nums, want",
    (
        ([1, 3, 4, 2, 2], 2),
        ([3, 1, 3, 4, 2], 3),
    ),
)
def test_findDuplicate(fn: Callable, nums: List[int], want: int):
    got = fn(nums)
    assert got == want

File: c/src/findDuplicate_test.c
#include <assert.h>
#include <stdlib.h>

#include "findDuplicate.c"

typedef struct test {
    int* nums;
    int numsSize;
    int want;
} test;

int main(int argc, char* argv[])
{
    test tests[] = {
        { (int[]) { 1, 3, 4, 2, 2 }, 5, 2 },
        { (int[]) { 3, 1, 3, 4, 2 }, 5, 3 },
    };
    size_t len_tests = sizeof(tests) / sizeof(test);

    for (size_t i = 0; i < len_tests; i++) {
        test* t = &tests[i];
        int got = findDuplicate(t->nums, t->numsSize);
        assert(got == t->want);
    }

    exit(EXIT_SUCCESS);
}
